### PR TITLE
Optional parent tracking

### LINKED ISSUE #167: Optional tracking of parent functionality
Hey, thank you for this lovely gem.

While integrating it in one codebase we had an exception coming from this line https://github.com/DmitryTsepelev/store_model/blob/085ee5e9d9720a45a1c665dc7de6e4015ddb079b/lib/store_model/ext/parent_assignment.rb#L12

The reason for it was that there was an Active Record class backed by table which had a `method` column. Basically #121 and #122.
We could easily workaround that, but it made me aware of something that I did not expect while reading the README. That this gem [extends](https://github.com/DmitryTsepelev/store_model/blob/master/lib/store_model/railtie.rb) ActiveRecord and ActiveModel functionality quite deeply by extending getter and setter behavior. 

As far as I checked by reading the code and running specs, this is only needed for parent tracking #35. We don't need this functionality and would prefer minimizing the risk of affecting rails behavior, especially in models that don't use StoreModel.

Would you be open to a PR which disables parent tracking functionality based on a configuration option?

I imagine it would work like

```
StoreModel.config.enable_parent_assignment = false # default would remain to be true
```

and then 

```
config.to_prepare do |_app|
  ActiveSupport.on_load(:active_record) do
    if StoreModel.config.enable_parent_assignment
      ActiveModel::Attributes.prepend(Attributes)
      prepend(Base)
    end
  end
end
```

I tested it in my app and the attribute value (set by the app) is available during the extending phase.

Since the behavior is global, to test this functionality I would probably create a second `dummy` rails app in store_model specs.

### PR DESCRIPTION
**Pull request by paneq**
* Optional parent tracking functionality. Can be disabled with settings.
* Docs updated.
* Specs changed to verify both approaches, including github actions workflow.

## TODO

* [x] Remove TODO from code once GH workflow is approved and verified to be working correctly (failing for ENABLE_PARENT_ASSIGNMENT=false now)

```
          # TODO: Remove me!
          expect("Verify correct github workflow").to eq("This should fail too see ENV ENABLE_PARENT_ASSIGNMENT working")
```

Fixes #167

### CODE REVIEW

File: spec/store_model/model_spec.rb [hunk 1]
```diff
@@ -177,12 +177,24 @@
...
     shared_examples "for singular type" do
       it "returns parent instance" do
-        expect(subject.configuration.parent).to eq(subject)
+        if StoreModel.config.enable_parent_assignment
+          expect(subject.configuration.parent).to eq(subject)
+        else
+          expect(subject.configuration.parent).to be_nil
+
+          # TODO: Remove me!
+          expect("Verify correct github workflow").to eq("This should fail too see ENV ENABLE_PARENT_ASSIGNMENT working")
```
Comment (paneq) [inline spec/store_model/model_spec.rb:186]: TODO: Remove me

**Comment by DmitryTsepelev** [comment]
Not passing for some reason

**Comment by paneq** [comment]
@DmitryTsepelev It's not passing because of an intentional error that I added to verify that the github actions are executing both versions (with and without tracking) and that the ENV assignment works.

```
  1) StoreModel::Model#parent json activerecord model parent returns parent instance
     Failure/Error: expect("Verify correct github workflow").to eq("This should fail too see ENV ENABLE_PARENT_ASSIGNMENT working")

       expected: "This should fail too see ENV ENABLE_PARENT_ASSIGNMENT working"
            got: "Verify correct github workflow"
```

File: spec/dummy/config/initializers/enable_parent_assignment.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+if ENV.key?("ENABLE_PARENT_ASSIGNMENT")
+  enable_parent_assignment = case ENV.fetch("ENABLE_PARENT_ASSIGNMENT")
```
[the remark this answers is not in the document]
Comment (paneq) [reply spec/dummy/config/initializers/enable_parent_assignment.rb:2]: Added `.strip.downcase`

**Review by DmitryTsepelev — APPROVED** [review]
LGTM, thank you so much! ☺️

**Pull request merged**